tests/functional_tests/ffront_tests/test_unrollPowOp.py

```diff
@@ -0,0 +1,176 @@
+import pathlib
+
+import pytest
+
+from eve.pattern_matching import ObjectPattern
+from functional.ffront import common_types as ct, field_operator_ast as foast
+from functional.ffront.foast_passes.type_deduction import FieldOperatorTypeDeductionError
+from functional.ffront.foast_passes.unroll_power_op import UnrollPowerOp
+
+
+def _make_power_testee(pow_n: int) -> foast.BinOp:
+    loc = foast.SourceLocation(line=106, column=12, source=str(pathlib.Path(__file__).resolve()))
+
+    testee = foast.BinOp(
+        right=foast.Constant(
+            dtype=ct.ScalarType(kind=ct.ScalarKind.INT),
+            location=loc,
+            value=str(pow_n),
+            type=ct.ScalarType(kind=ct.ScalarKind.INT),
+        ),
+        left=foast.Name(
+            id="a",
+            type=ct.DeferredSymbolType(constraint=None),
+            location=loc,
+        ),
+        op=foast.BinaryOperator.POW,
+        location=loc,
+        type=ct.DeferredSymbolType(constraint=None),
+    )
+    return testee
+
+
+power_1 = ObjectPattern(foast.Name, id="a")
+power_2 = ObjectPattern(
+    foast.BinOp, right=ObjectPattern(foast.Name, id="a"), op=foast.BinaryOperator.MULT
+)
+power_3 = ObjectPattern(
+    foast.BinOp,
+    right=ObjectPattern(foast.Name, id="a"),
+    left=ObjectPattern(
+        foast.BinOp,
+        right=ObjectPattern(foast.Name, id="a"),
+        left=ObjectPattern(foast.Name, id="a"),
+        op=foast.BinaryOperator.MULT,
+    ),
+    op=foast.BinaryOperator.MULT,
+)
```

Comment:
Please place the test data into a dedicated function to avoid polluting the module.
```suggestion
def power_test_cases():
  return (
    # exponent, expected
    (1, ObjectPattern(foast.Name, id="a")),
    (2, ObjectPattern(
        foast.BinOp, right=ObjectPattern(foast.Name, id="a"), op=foast.BinaryOperator.MULT
    )),
    (3, ObjectPattern(
        foast.BinOp,
        right=ObjectPattern(foast.Name, id="a"),
        left=ObjectPattern(
            foast.BinOp,
            right=ObjectPattern(foast.Name, id="a"),
            left=ObjectPattern(foast.Name, id="a"),
            op=foast.BinaryOperator.MULT,
        ),
        op=foast.BinaryOperator.MULT,
    ))
  )
```